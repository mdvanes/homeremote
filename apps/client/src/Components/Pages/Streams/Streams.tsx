import React, { FC, useCallback, useEffect } from "react";
import HomeremoteStreamPlayer from "@mdworld/homeremote-stream-player";
import { styled } from "@mui/material/styles";
import { Container, lighten } from "@mui/system";

const StreamContainer = styled(Container)(
    ({ theme }) => `
    padding: 0 !important;
    font-family: 'Roboto';
    max-width: 800px;
    
    .card {
        background-color: ${lighten(theme.palette.background.paper, 2 * 0.025)};
    }

    .card .channel {
        flex-direction: column;
        align-items: flex-start;
    }

    .card .channel select {
        color: ${theme.palette.primary.main};
        border-bottom-color: ${theme.palette.primary.main};
    }

    .card .channel .channel-info {
        color: ${theme.palette.text.primary};
        max-height: 3rem;
    }

    .card .controls > .play {
        fill: ${theme.palette.text.primary};

        &:hover, &:hover.active {
            fill: ${theme.palette.primary.light};
        }

        &.active {
            fill: ${theme.palette.primary.main};
        }
    }

    .card .artist, .card .title {
        display: inline;
    }

    .card .artist {
        color: ${theme.palette.primary.main};
    }

    .card .title {
        font-size: 100%;
        margin-right: 0.5rem;
    }
`
);

const Streams: FC = () => {
    // handle what happens on key press
    const handleKeyPress = useCallback((event) => {
        if (event.key === "Control") {
            return;
        }
        if (event.ctrlKey) {
            console.log(`Key pressed: ctrl+${event.key}`);
        }
        // p for pause
        if (event.key === "t") {
            console.log("toggle radio v stream");
        }
    }, []);

    useEffect(() => {
        document.addEventListener("keydown", handleKeyPress);

        return () => {
            document.removeEventListener("keydown", handleKeyPress);
        };
    }, [handleKeyPress]);

    return (
        <StreamContainer>
            <HomeremoteStreamPlayer url={process.env.NX_BASE_URL || ""} />
        </StreamContainer>
    );
};

export default Streams;
