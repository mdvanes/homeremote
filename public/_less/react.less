// for /react test

// Toggle
.btn-icon {
    font-size: 350%;
}

// Button-Set
.btn-group-justified {
    > a,
    > span {
        font-size: 230%;
    }
}

.margin-top {
    margin-top: 1em;
}

.log {
    color: #fff;
    height: 300px;
}