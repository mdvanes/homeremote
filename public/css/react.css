.btn-icon {
  font-size: 350%;
}
.btn-group-justified > a,
.btn-group-justified > span {
  font-size: 230%;
}
.margin-top {
  margin-top: 1em;
}
.log {
  color: #fff;
  height: 300px;
}
