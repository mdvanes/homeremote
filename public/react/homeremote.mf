CACHE MANIFEST
index.html
../library/bootstrap/css/bootstrap.min.css
../library/bootstrap/fonts/glyphicons-halflings-regular.woff2
../library/bootstrap/fonts/glyphicons-halflings-regular.woff
../library/bootstrap/fonts/glyphicons-halflings-regular.ttf
../css/react.css
build/bundle.js

NETWORK:
*
