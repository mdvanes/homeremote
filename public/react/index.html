<!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>React</title>
    
    
  </head>
  <body>
    
    <script src="build/bundle.js"></script>
    
  </body>
</html>
