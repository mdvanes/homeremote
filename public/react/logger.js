/**
 * Created by m.van.es on 6-2-2016.
 */

let logger = {};

let writeLog = (...messages) => {
    let time = new Date();
    let timestamp = time.getHours() + ':' +
        time.getMinutes() + '.' +
        time.getSeconds();
    document.querySelector('.log').value = timestamp + ' ' +
        messages.join(' ') + '\n' +
        document.querySelector('.log').value;
};

logger.log = message => {
    writeLog('INFO:    ', message);
};

logger.error = message => {
    writeLog('ERROR:', message);
};

export default logger;