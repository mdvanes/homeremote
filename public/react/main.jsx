// from https://github.com/code0wl/react-example-es2015/blob/master/app/main.jsx#L5
//import './stylesheets/main.css';
import React from 'react';
import {render} from 'react-dom';
////import FilterableProductTable from './src/filterable-product-table';
import ButtonGroup from './components/button-group';
import Toggle from './components/toggle';
import Log from './components/log';

// init shell
renderShell();

// TODO toggle as single button with icon (instead of 2 buttons)

function renderShell() {
    var shell = document.createElement('div');
    shell.className = 'app-shell container';
    document.body.appendChild(shell);
    render(
        <div>
            <div className="row">
                <div className="col-xs-12">
                    <ButtonGroup label=" 1" icon="lamp" id="clickstub"/>
                    <ButtonGroup label=" 2" icon="lamp" id="clickstub"/>
                    <ButtonGroup label=" 3" icon="lamp" id="clickstub"/>
                    <ButtonGroup label=" 4" icon="lamp" id="clickstub"/>
                </div>
            </div>
            <div className="row margin-top">
                {/*<Toggle label=" 2" id="togglestub" icon="lamp"/>
                <Toggle label=" 3" id="togglestub" icon="lamp"/>
                <Toggle label="Switch 4" id="togglestub"/>*/}
                <Toggle label="" id="togglestub" icon="volume-up"/>{/* Broadcast */}
                <Toggle label="" id="togglestub" icon="music"/>{/* Music on server */}
            </div>
            <div className="row margin-top">
                <div className="col-xs-12">
                    <Log/>
                </div>
            </div>
        </div>, shell);
}

