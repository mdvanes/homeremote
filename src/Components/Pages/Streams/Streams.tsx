import React, { FC } from "react";
import HomeremoteStreamPlayer from "@mdworld/homeremote-stream-player";

const Streams: FC = () => (
    <div style={{ fontFamily: "Roboto" }}>
        <HomeremoteStreamPlayer url={process.env.REACT_APP_BASE_URL || ""} />
    </div>
);

export default Streams;
