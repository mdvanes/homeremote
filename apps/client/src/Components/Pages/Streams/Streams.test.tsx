import React from "react";
import { render } from "@testing-library/react";
import Streams from "./Streams";

jest.mock("@mdworld/homeremote-stream-player", () => "mock-stream-player");
jest.mock("../../Molecules/Jukebox/Jukebox", () => "mock-jukebox");

describe("Streams page", () => {
    it("contains the stream player", () => {
        const { baseElement } = render(<Streams />);
        expect(baseElement).toMatchSnapshot();
    });
});
