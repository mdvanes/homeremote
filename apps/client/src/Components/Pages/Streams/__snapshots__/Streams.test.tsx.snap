// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Streams page contains the stream player 1`] = `
<body>
  <div>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-d93ppg-MuiContainer-root"
    >
      <mock-stream-player
        url="http://localhost"
      />
    </div>
    <div
      class="MuiBox-root css-1yuhvjn"
    />
    <mock-jukebox />
  </div>
</body>
`;
